extensions.bzl

```diff
@@ -0,0 +1,44 @@
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+load("@bazel_tools//tools/build_defs/repo:utils.bzl", "maybe")
+load(":deps.bzl", "bazel_bats_dependencies")
+
+def _bazel_bats_deps_impl(ctx):
+    bats_core_version = None
+    bats_core_sha256 = None
+    bats_assert_version = None
+    bats_assert_sha256 = None
+    bats_support_version = None
+    bats_support_sha256 = None
+    for mod in ctx.modules:
```

Comment:
Just a drive-by comment:

this resolution logic might be strange because [ctx.modules](https://bazel.build/rules/lib/builtins/module_ctx?hl=en#modules) returns the modules in BFS order from the root project, this means the root version of bats tools will be overridden by its farthest dependency.

Reply:
@Wyverald @fmeum Do we have some MVS util functions like this somewhere?

Reply:
A simple approach would be to use `module.is_root` to fail on non-root module tags. Modules can use the extension with `dev_dependency = True` to set the version for their own tests while remaining usable as a dependency.

Neither skylib nor bazel-lib have a semver module at this point, but this would be a generally useful addition. https://github.com/bazel-contrib/bazel-gazelle/blob/master/internal/bzlmod/semver.bzl is pretty simple and could serve as a base.

Reply:
do note that Bazel module versions are a superset of semver, so Bazel itself offering some sort of semver (as opposed to bzlmod version) utils would be a bit weird.

OTOH, for this specific use case, you could just use first-wins (instead of what's currently here, which is last-wins). First-wins is the semantics for toolchain resolution as well, so at least there's precedent.

Reply:
Noted. At the moment it doesn't really matter as there is the only one version that is released but I will revisit it upon any next release.